guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+for later training. The utility can be found at
+https://github.com/keras-team/keras-cv/blob/master/examples/training/object_detection_3d/waymo/serialize_records.py
+
+Next up, let's augment our data! In partnership with Waymo, KerasCV offers a
+set of state-of-the-art 3D augmentations for LIDAR data and 3D boxes. They
+behave like all Keras preprocessing layers, and they're very easy to set up.
+"""
+
+augmentations = keras.Sequential([
+  keras_cv.layers.GlobalRandomFlip(),
+  keras_cv.layers.GlobalRandomRotation(max_rotation_angle_z=3.14),
+])
+
+dataset = dataset.map(augmentations)
+
+"""
+In just a few lines of code, we've augmented our input data using a few of the
+3D augmentations offered in KerasCV. Check out the whole set of augmentations
+at https://github.com/keras-team/keras-cv/tree/master/keras_cv/layers/preprocessing_3d.
+
+Next, we'll create a `MultiHeadCenterPillar` model to train. These models are
+very configurable, and the configuration can be a bit overwhelming at first.
+So let's start by defining (and explaining!) some of the configuration.
+
+For a more in-depth understanding of how the model works, check out
+https://arxiv.org/pdf/2006.11275.pdf.
+"""
+
+"""
+Our model will group points into voxels in 3D space, and we need to specify
+how large these voxels will be. Here, we define the width, length, and height
+of each voxel in the units used by the input data (meters, in the case of
+Waymo Open Dataset).
+
+Because we're predicting vertical boxes, it's common to use arbitrarily tall
+voxels, so in this case we use 1000 for the z dimension.
+"""
+voxel_size = [0.32, 0.32, 1000]
+
+"""
+For voxelization, we also need to specify the global volume of our voxel space,
+which represents the overall target area where we will identify boxes. Here
+we use a range of -256 * voxel_size to 256 * voxel_size for the x and y
+size, and -20 to 20 for the z size. As a result, we will produce voxel features
+in an overall grid of 512x512x1 voxels.
+"""
+# 81.92 = 256 * 0.32
+spatial_size = [-81.92, 81.92, -81.92, 81.92, -20, 20]
+
+"""
+After voxelizing points, we'll run the results through a point net, which is
+a dense network with a configurable feature size. Here we define this feature
+size.
+"""
+voxelization_feature_size = 128
+
+"""
+We'll also want to know a prior for the length, width, and height of each of
+the classes we're trying to detect. This is somewhat akin to the concept of
+anchor sizes in 2D object detection, but is used for numerical regularization
+instead of prediction anchoring in this case.
+"""
+car_anchor_size = [4.5, 2.0, 1.6]
+pedestrian_anchor_size = [0.6, 0.8, 1.8]
+
+"""
+Now we can build our model!
+
+We'll define a function to create the model so that we can initialize it inside
+of a tf.distribute scope later on.
+"""
+
+
+def build_centerpillar_model():
+  """
+  Our first model component is a voxelization layer. This will be used to
```

Comment:
its a shame we can't just show this off in MD blocks- but I see why.

maybe we do it once, then we re-define the function below?